test/functionalTest/cases/2877_return_ServicePath_in_entity_response/servicePath_behave_with_q_and_orderBy.test

```diff
@@ -0,0 +1,218 @@
+echo
+echo
+
+
+echo "02. Create entity E2 with temperature = 25 in servicePath /Barcelona/B"
+echo "======================================================================"
+payload='{
+  "id": "E2",
+  "type": "T",
+  "temperature": {
+    "value": 25,
+    "type": "Integer"
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload" --servicePath /Barcelon/B
+echo
+echo
+
+
+echo "03. Create entity E3 with temperature = 30 in servicePath /Madrid/A"
+echo "==================================================================="
+payload='{
+  "id": "E3",
+  "type": "T",
+  "temperature": {
+    "value": 30,
+    "type": "Integer"
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload" --servicePath /Madrid/A
+echo
+echo
+
+
+echo "04. Create entity E4 with temperature = 40 in servicePath /Barcelona/B"
+echo "======================================================================"
+payload='{
+  "id": "E4",
+  "type": "T",
+  "temperature": {
+    "value": 40,
+    "type": "Integer"
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload" --servicePath /Barcelona/B
+echo
+echo
+
+
+echo "05. GET entities with q parameter and servicePath /Madrid/A"
+echo "==========================================================="
+orionCurl --url /v2/entities?q=servicePath:/Madrid/A
+echo
+echo
+
+
+echo "06. GET entities with orderBy servicePath"
+echo "========================================="
+orionCurl --url /v2/entities?orderBy=servicePath
+echo
+echo
+
+
+--REGEXPECT--
+01. Create entity E1 with temperature = 35 in servicePath /Madrid/A
+===================================================================
+HTTP/1.1 201 Created
+Date: REGEX(.*)
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Location: /v2/entities/E1?type=T
+Content-Length: 0
+
+
+
+02. Create entity E2 with temperature = 25 in servicePath /Barcelona/B
+======================================================================
+HTTP/1.1 201 Created
+Date: REGEX(.*)
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Location: /v2/entities/E2?type=T
+Content-Length: 0
+
+
+
+03. Create entity E3 with temperature = 30 in servicePath /Madrid/A
+===================================================================
+HTTP/1.1 201 Created
+Date: REGEX(.*)
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Location: /v2/entities/E3?type=T
+Content-Length: 0
+
+
+
+04. Create entity E4 with temperature = 40 in servicePath /Barcelona/B
+======================================================================
+HTTP/1.1 201 Created
+Date: REGEX(.*)
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Location: /v2/entities/E4?type=T
+Content-Length: 0
+
+
+
+05. GET entities with q parameter and servicePath /Madrid/A
+===========================================================
+HTTP/1.1 200 OK
+Date: REGEX(.*)
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Content-Type: application/json
+Content-Length: 2
+
+[]
+
+
+06. GET entities with orderBy servicePath
+=========================================
```

Comment:
It seems orderBy is not working. E2 and E4 should bome before E1 and E3.

Maybe it's a matter of fixing sortCriteria() function.

In addition, for completeness plaease include a step for `GET /v2/entities?orderBy=!servicePath`

Reply:
Fixed and added in 9e92496.